Azure Deploy Targets refresh (#2593)
OctopusDeploy/docs: src/pages/docs/infrastructure/deployment-targets/azure/index.md

- [Azure Service Fabric Clusters](/docs/infrastructure/deployment-targets/azure/service-fabric-cluster-targets).
- [Azure Web Apps](/docs/infrastructure/deployment-targets/azure/web-app-targets) (also works for Azure Functions).
- Azure Kubernetes Service via the [Kubernetes Agent](/docs/kubernetes/targets/kubernetes-agent) and [Kubernetes API](/docs/kubernetes/targets/kubernetes-api) deployment targets.
- Azure VM via [Tentacle using Desired State Configuration (DSC)](/docs/infrastructure/deployment-targets/tentacle/windows/azure-virtual-machines/via-an-arm-template-with-dsc).

:::div{.warning}

> Octopus Deploy versions prior to `2024.1` deploying to Azure Cloud Services was supported([See deprecation notice for more details](/docs/deprecations#azure-cloud-services-classic)) but as of September 1st 2024 Azure Cloud Services(Classic) Microsoft has deprecated these services and as of October 1st 2024 Microsoft is shutting down any existing running Cloud Service deployments. ([Source](https://learn.microsoft.com/en-us/azure/cloud-services/cloud-services-choose-me))

Reviewer: ```suggestion
Azure Cloud Services are no longer supported in Octopus Deploy as of `2025.1`. 

Microsoft has deprecated these Azure services, and as of October 1st 2024 shut down existing Cloud Service deployments. ([Source](https://learn.microsoft.com/en-us/azure/cloud-services/cloud-services-choose-me))
```
